From 6d2c2e2cab8a3fc963f57764212b59ac0046d594 Mon Sep 17 00:00:00 2001
From: Robert Chiras <robert.chiras@nxp.com>
Date: Thu, 9 Nov 2017 14:05:53 +0200
Subject: [PATCH 2864/5242] MLK-16918-9: arm64: dts: fsl-imx8qm-lpddr4-arm2:
 Update dts files for nwl-mipi-dsi bridge

commit  2a91dfe0b5d7bcbf081a6fd216a909f5f9b4d8d9 from
https://source.codeaurora.org/external/imx/linux-imx.git

Now, the NWL MIPI-DSI driver is implemented as a real bridge and uses
the "nwl,mipi-dsi" compatible.
This patch updates the ADV7535 and RM67191 specific DTS files to 8QM.

Signed-off-by: Robert Chiras <robert.chiras@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../fsl-imx8qm-lpddr4-arm2-dsi-adv7535.dts         |   28 +++++++++-----------
 .../fsl-imx8qm-lpddr4-arm2-dsi-rm67191.dts         |   16 ++++++++---
 2 files changed, 24 insertions(+), 20 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2-dsi-adv7535.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2-dsi-adv7535.dts
index 8793030..3ef6a51 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2-dsi-adv7535.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2-dsi-adv7535.dts
@@ -14,10 +14,6 @@
 
 #include "fsl-imx8qm-lpddr4-arm2.dts"
 
-&hdmi {
-	status = "disabled";
-};
-
 &ldb1_phy {
 	status = "disabled";
 };
@@ -26,14 +22,6 @@
 	status = "disabled";
 };
 
-&ldb2_phy {
-	status = "disabled";
-};
-
-&ldb2 {
-	status = "disabled";
-};
-
 &i2c0_mipi_dsi0 {
 	#address-cells = <1>;
 	#size-cells = <0>;
@@ -50,7 +38,7 @@
 
 		port {
 			adv7535_1_in: endpoint {
-				remote-endpoint = <&mipi_dsi1_out>;
+				remote-endpoint = <&mipi_dsi_bridge1_adv>;
 			};
 		};
 	};
@@ -62,9 +50,13 @@
 
 &mipi_dsi1 {
 	status = "okay";
+};
+
+&mipi_dsi_bridge1 {
+	status = "okay";
 
 	port@1 {
-		mipi_dsi1_out: endpoint {
+		mipi_dsi_bridge1_adv: endpoint {
 			remote-endpoint = <&adv7535_1_in>;
 		};
 	};
@@ -86,7 +78,7 @@
 
 		port {
 			adv7535_2_in: endpoint {
-				remote-endpoint = <&mipi_dsi2_out>;
+				remote-endpoint = <&mipi_dsi_bridge2_adv>;
 			};
 		};
 	};
@@ -98,9 +90,13 @@
 
 &mipi_dsi2 {
 	status = "okay";
+};
+
+&mipi_dsi_bridge2 {
+	status = "okay";
 
 	port@1 {
-		mipi_dsi2_out: endpoint {
+		mipi_dsi_bridge2_adv: endpoint {
 			remote-endpoint = <&adv7535_2_in>;
 		};
 	};
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2-dsi-rm67191.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2-dsi-rm67191.dts
index fee12537..aa25d9b 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2-dsi-rm67191.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2-dsi-rm67191.dts
@@ -40,6 +40,10 @@
 
 &mipi_dsi1 {
 	status = "okay";
+};
+
+&mipi_dsi_bridge1 {
+	status = "okay";
 
 	panel@0 {
 		compatible = "raydium,rm67191";
@@ -51,13 +55,13 @@
 		panel-height-mm = <121>;
 		port {
 			panel1_in: endpoint {
-				remote-endpoint = <&mipi1_out>;
+				remote-endpoint = <&mipi_bridge1_out>;
 			};
 		};
 	};
 
 	port@1 {
-		mipi1_out: endpoint {
+		mipi_bridge1_out: endpoint {
 			remote-endpoint = <&panel1_in>;
 		};
 	};
@@ -69,6 +73,10 @@
 
 &mipi_dsi2 {
 	status = "okay";
+};
+
+&mipi_dsi_bridge2 {
+	status = "okay";
 
 	panel@0 {
 		compatible = "raydium,rm67191";
@@ -80,13 +88,13 @@
 		panel-height-mm = <121>;
 		port {
 			panel2_in: endpoint {
-				remote-endpoint = <&mipi2_out>;
+				remote-endpoint = <&mipi_bridge2_out>;
 			};
 		};
 	};
 
 	port@1 {
-		mipi2_out: endpoint {
+		mipi_bridge2_out: endpoint {
 			remote-endpoint = <&panel2_in>;
 		};
 	};
-- 
1.7.9.5

